lib/reportData.js

```diff
@@ -58,30 +63,40 @@ var objectToQueryString = function (params) {
  * @returns {String}
  */
 var prepareURL = function (url) {
-	if (/\?/.test(url)) {
-		if (url.charAt(url.length - 1) !== '&') {
-			url += '&';
-		}
-	} else {
-		url += '?';
-	}
-	return url;
+    if (/\?/.test(url)) {
+        if (url.charAt(url.length - 1) !== '&') {
+            url += '&';
+        }
+    } else {
+        url += '?';
+    }
+    return url;
 };
 
-var sendReport = function (url, isUnloadEvent) {
-
-	// TODO use navigator.sendBeacon for async logging calls once cross browser
-	// support and its limitations (see: CORS restricted) can be indentified
-
-	var wn = window.navigator;
-	if (isUnloadEvent && wn.userAgent.toLowerCase().indexOf('safari') > -1) {
-		xhrTrack(url);
-	} else {
-		imgTrack(url);
-	}
-
+/**
+ *
+ * @param url
+ * @param isUnloadEvent
+ * @param {Boolean} [isSendBeacon]
+ * @param {String} [sendBeaconJsonString]
```

Comment:
We should update this argument so that what's currently called 'sendBeaconJsonString' can be an object or a string; if it's an object, then we just run JSON.stringify() on it which will avoid users having to do that every time they call this function.